### PR TITLE
docs: add hardhat example + ci

### PR DESCRIPTION
**Pull request by platocrat**
**Description**
Replaces #501 with a single PR for the `optimism-tutorial`. 
Additional PRs to follow for the following repositories:
1. [Waffle-ERC20-Example](https://github.com/ethereum-optimism/Waffle-ERC20-Example)
2. [Truffle-ERC20-Example](https://github.com/ethereum-optimism/Truffle-ERC20-Example)
3. [l1-l2-deposit-withdrawal](https://github.com/ethereum-optimism/l1-l2-deposit-withdrawal)

### CODE REVIEW

**Comment by gakonst** [comment]
I've renamed the package to `hardhat-example` (better than calling it "optimism tutorial", too generic), since it'll be in line with the rest of the examples (waffle and truffle)

**Comment by gakonst** [comment]
Converted to draft, to merge after figuring out the bug in https://github.com/ethereum-optimism/optimism/pull/646.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -42,3 +43,16 @@ jobs:
             yarn build:contracts
             yarn build:contracts:ovm
             yarn test:integration
+
+      # Examples Tests
+      - name: Test & deploy hardhat-example on hardhat (regression)
+        working-directory: ./examples/hardhat
+        run: |
+          yarn deploy
```
Comment (platocrat) [inline .github/workflows/integration.yml:51]: I noticed you are deploying before testing. Is this intentional?

Comment (platocrat) [reply .github/workflows/integration.yml:51]: you also never compile the contracts with `yarn compile`

Comment (gakonst) [reply .github/workflows/integration.yml:58]: Compiling happens automatically by hardhat, so no need for us to do it explicitly here

**Pull request merged**